tests/virtual_machines_test.go

```diff
@@ -69,11 +69,42 @@ var _ = Describe("Virtual Machines", func() {
 				return len(vmList.Items)
 
 			}, timeout, pollingInterval).Should(Equal(0), "failed to remove all vm objects")
+
+			for _, namespace := range []string{TestNamespace, OtherTestNamespace} {
+				err = cleanNamespaceLabels(namespace)
+				Expect(err).ToNot(HaveOccurred())
+			}
+		})
+
+		Context("When the client wants to create a vm on an opted-out namespace", func() {
+			It("should create a vm object without kubemacpool MAC address assignment when no label is set to namespace", func() {
+				err := initKubemacpoolParams(rangeStart, rangeEnd, map[string]string{})
```

Comment:
This should be in a BeforeEach of the `Context` to set up the context. I know it is not the case in the rest of the module, but let's take this opportunity to write it properly.

Reply:
Not sure I understand. 
The reason these parameters are not in BeforeEach is that they change on *each* test.
please elaborate

Reply:
ok I think I got what you mean